api/v1beta1/conversion_test.go

```diff
@@ -21,11 +21,53 @@ import (
 
 	. "github.com/onsi/gomega"
 
-	runtime "k8s.io/apimachinery/pkg/runtime"
-	v1beta2 "sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
+	fuzz "github.com/google/gofuzz"
+	"k8s.io/apimachinery/pkg/api/apitesting/fuzzer"
+	"k8s.io/apimachinery/pkg/runtime"
+	runtimeserializer "k8s.io/apimachinery/pkg/runtime/serializer"
+	"sigs.k8s.io/cluster-api-provider-aws/v2/api/v1beta2"
 	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
 )
 
+func fuzzFuncs(_ runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		AWSMachineFuzzer,
+		AWSMachineTemplateFuzzer,
+	}
+}
+
+func AWSMachineFuzzer(obj *AWSMachine, c fuzz.Continue) {
+	c.FuzzNoCustom(obj)
+	
+	// AWSMachine.Spec.FailureDomain, AWSMachine.Spec.Subnet.ARN and AWSMachine.Spec.AdditionalSecurityGroups.ARN has been removed in v1beta2, so setting it to nil in order to avoid v1beta1 --> v1beta2 --> v1beta1 round trip errors.
```

Comment:
I'm not sure setting them to nil is teh right way to do the v1beta1 --> v1beta2 --> v1beta1.

A user may have FailureDomains/ARN set in v1beta1, so i think the tests should store/restore the values. wdyt?

Reply:
Doesn't feel right but it's what's done. I'd have though there was a pattern to preserve this. So we can resolve this.